components/script/dom/bindings/trace.rs

```diff
@@ -160,6 +160,50 @@ impl<T: JSTraceable> JSTraceable for Rc<T> {
     }
 }
 
+impl<T: JSTraceable> JSTraceable for Arc<T> {
+    fn trace(&self, trc: *mut JSTracer) {
+        (**self).trace(trc)
+    }
+}
+
+impl JSTraceable for Mutex<Option<SharedRt>> {
```

Comment:
feel like we should be using the macro for these so that we can later lint that the impl is on a type that exists completely outside of this crate